WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/WooShipping Create Shipping Labels/WooShipping Split Shipments/SelectableShipmentRow.swift

```diff
@@ -0,0 +1,96 @@
+import Yosemite
+import SwiftUI
+
+/// Row for a selectable shipment item to ship with the Woo Shipping extension.
+struct SelectableShipmentRow: View {
+    @ObservedObject private var viewModel: SelectableShipmentRowViewModel
+
+    init(viewModel: SelectableShipmentRowViewModel) {
+        self.viewModel = viewModel
+    }
+
+    @ScaledMetric private var scale: CGFloat = 1
+
+    var body: some View {
+        AdaptiveStack(spacing: Layout.horizontalSpacing) {
+            if viewModel.isSelectable {
+                selectionCircle(selected: viewModel.selected)
+                    .contentShape(Rectangle())
+                    .onTapGesture {
+                        viewModel.handleTap()
+                    }
+            }
+
+            ProductImageThumbnail(productImageURL: viewModel.item.imageUrl,
+                                  productImageSize: Layout.imageSize,
+                                  scale: scale,
+                                  productImageCornerRadius: Layout.imageCornerRadius,
+                                  foregroundColor: Color(UIColor.listSmallIcon))
+            .overlay(alignment: .topTrailing) {
+                if viewModel.showQuantity {
+                    BadgeView(text: viewModel.item.quantityLabel,
+                              customizations: .init(textColor: .white, backgroundColor: .black),
+                              backgroundShape: badgeStyle)
+                    .offset(x: Layout.badgeOffset, y: -Layout.badgeOffset)
```

Comment:
Nit: It looks to me that the badge is missing a white border around it when the label is longer.

<img width="115" alt="image" src="https://github.com/user-attachments/assets/bc6e4b3a-12b1-4ad2-bcda-3c2be1b815a9" />

Reply:
I think this can be quickly fixed by adding `stroke` to the background view of the `BadgeView` with style `roundedRectangle`. But fixing it in a separate PR sounds good to me too.